From 8f05671597898ffc9f2f310bbf71e0b9c7b4dec3 Mon Sep 17 00:00:00 2001
From: Crystal Wood <crwood@redhat.com>
Date: Mon, 22 Jan 2024 16:13:20 -0600
Subject: [PATCH 2/3] rt-tests: cyclicdeadline: Print the histogram regardless
 of quiet

The histogram printing code should not have been gated by !quiet,
even though other summary printing code is.

The non-histogram output also should not have been gated based on
the presence of the histogram.

Signed-off-by: Crystal Wood <crwood@redhat.com>
Signed-off-by: John Kacur <jkacur@redhat.com>
---
 src/sched_deadline/cyclicdeadline.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/sched_deadline/cyclicdeadline.c b/src/sched_deadline/cyclicdeadline.c
index b3155547b9bb..3cb8f714b788 100644
--- a/src/sched_deadline/cyclicdeadline.c
+++ b/src/sched_deadline/cyclicdeadline.c
@@ -1117,15 +1117,15 @@ static void loop(struct sched_data *sched_data, int nr_threads)
 	usleep(10000);
 	if (!quiet) {
 		printf("\033[%dB", nr_threads + 2);
-	} else {
-		if (histogram) {
-			FILE *out = histfile ? histfile : stdout;
+	} else if (!histogram) {
+		for (i = 0; i < nr_threads; ++i)
+			print_stat(stdout, &sched_data[i], i, 0, 0);
+	}
 
-			print_hist(out, sched_data, nr_threads);
-		} else {
-			for (i = 0; i < nr_threads; ++i)
-				print_stat(stdout, &sched_data[i], i, 0, 0);
-		}
+	if (histogram) {
+		FILE *out = histfile ? histfile : stdout;
+
+		print_hist(out, sched_data, nr_threads);
 	}
 }
 
-- 
2.43.0

